site-src/guides/infrastructure.md

```diff
@@ -0,0 +1,45 @@
+# Gateway infrastructure labels and annotations
+
+??? success "Standard Channel since v1.2.0"
+
+    The `infrastructure` field is GA and has been part of the Standard Channel since
+    `v1.2.0`. For more information on release channels, refer to our [versioning
+    guide](/concepts/versioning).
+
+Gateway API implementations are responsible for creating the backing
+infrastructure needed to make each Gateway work. For example, implementations
+running in a Kubernetes cluster often create [Services][service] and
+[Deployments][deployment], while cloud-based implementations may create cloud
+load balancer resources. In many cases, it can be helpful to be able to
+propagate labels or annotations to these generated resources.
+
+
+The [`infrastructure` field][infrastructure] on a Gateway allows you to specify
+labels and annotations for the infrastructure created by the Gateway API controller.
+For example, if your Gateway infrastructure is running in-cluster, you can specify
+both Linkerd and Istio injection using the following Gateway configuration, making
+it simpler for the infrastructure to be incorporated into whichever service mesh
+you've installed.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: meshed-gateway
+  namespace: incoming
+spec:
+  gatewayClassName: meshed-gateway-class
+  listeners:
+  - name: http-listener
+    protocol: HTTP
+    port: 80
+  infrastructure:
+    labels:
+      istio-injection: enabled
```

Comment:
We should try to discourage using private labels. When the key doesn't have a DNS name then a `/` then a bit more key, it's cluster-private.

Reply:
Even if I generally agree with your statement, I think this is off-topic with this specific PR. This is an implementation-related problem, and we are not advertising any specific usage of labels here, but just showcasing some real-world examples.

Reply:
Across _Kubernetes_ as a project, we also like to promote good practise. We like that (promoting good practise) even if vendors ignore our nudges.

Reply:
If we could pick a real world example that avoids tacitly promoting bad practice, we should.